xcell/cls/cl.py

```diff
@@ -232,9 +232,52 @@ def __init__(self, data, tr1, tr2):
         self.outdir = self.get_outdir('fiducial')
         os.makedirs(self.outdir, exist_ok=True)
         self.cosmo = self.get_cosmo_ccl()
+        self.hm_par = self._get_halomodel_params()
         self._ccl_tr1 = None
         self._ccl_tr2 = None
 
+    def _get_halomodel_params(self):
+        if 'halo_model' not in self.data.data['cov']:
+            self.data.data['cov']['halo_model'] = {}
```

Comment:
What would you think of moving `halo_model` inside `fiducial` together with `cosmo`, `gc_bias`, etc.? My argument is that the halo model is not a property of the covariance, but of the modelling of the power spectrum

Reply:
Good idea! Done